lib/src/string_matchers.dart

```diff
@@ -122,8 +122,9 @@ class _StringContainsInOrder extends FeatureMatcher<String> {
   bool typedMatches(dynamic item, Map matchState) {
     var fromIndex = 0;
     for (var s in _substrings) {
-      fromIndex = item.indexOf(s, fromIndex);
-      if (fromIndex < 0) return false;
+      if (s.isEmpty) continue;
+      fromIndex = item.indexOf(s, fromIndex) + 1;
```

Comment:
This should actually be this probably right?

```suggestion
      fromIndex = item.indexOf(s, fromIndex) + s.length;
```

Reply:
Oh except that wouldn't work when you get `-1`. But the currently implementation also isn't correct, for instance I think we should have this test:

```
shouldFail('foo', stringContainsInOrder(['fo', 'oo']));
```

Which I think would (unexpectedly) pass in the current implementation?

Reply:
Ah true. Good catch! Will fix and push update soon!